cmapPy/math/robust_zscore.py

```diff
@@ -0,0 +1,31 @@
+'''
+robust_zscore.py
+
+Given a pandas df, and an optional control df, will calculate zscores using plate control or vehicle control
+'''
+
+def calc_zscore(mat, ctrl_mat=None):
+    '''
+    :param mat (pandas df): Matrix of data that zscoring will be applied to
+    :param ctrl_mat (pandas df): Optional subset matrix from which to draw medians and MADS (vehicle control)
+    :return zscore_data (pandas_df):
+    '''
+
+    # If optional df exists, calc medians and mads from it
+    if ctrl_mat is not None:
+        medians = ctrl_mat.median(axis=1)
+        median_devs = abs(ctrl_mat.subtract(medians, axis=0))
```

Comment:
Could use this: https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.mad.html

Reply:
^ Confused. Evan, I think Lev's suggesting you should just use the built-in median absolute deviation calculation...?

Reply:
The issue is that, as far as I can tell, there is no built in median absolute deviation calc. The one that Lev linked is mean absolute deviation